runtime/taskcontext.go

```diff
@@ -68,13 +76,17 @@ type TaskContextController struct {
 }
 
 // NewTaskContext creates a TaskContext and associated TaskContextController
-func NewTaskContext(tempLogFile string) (*TaskContext, *TaskContextController, error) {
+func NewTaskContext(tempLogFile string, claim *queue.TaskClaimResponse) (*TaskContext, *TaskContextController, error) {
```

Comment:
Then it's more obvious when we write tests that needs to fake it... Meaning it's more obvious what they have to provide...